tests/unit/kegd/test_processor.py

```diff
@@ -0,0 +1,206 @@
+import unittest
+import os
+import kubedriver.kegd.model as kegd_model
+import tests.unit.kegd.test_kegd_files as test_kegd_files
+import inspect
+import tests.utils as testutils
+from unittest.mock import MagicMock
+from ignition.service.templating import Jinja2TemplatingService
+from kubedriver.resourcedriver import ExtendedResourceTemplateContext, NameManager
+from kubedriver.kegd.manager import KegdStrategyLocationManager
+from kubedriver.kegd.processor import KegdStrategyLocationProcessor
+from kubedriver.kegd.properties import KegDeploymentProperties
+from kubedriver.kegd.strategy_files import KegDeploymentStrategyFiles
+from kubedriver.kegd.jobs import ProcessStrategyJob
+from kubedriver.kegd.model.strategy_execution import StrategyExecution, TaskGroup
+
+
+test_kegd_files_path = os.path.dirname(inspect.getfile(test_kegd_files))
+
+def get_kegd_files(name):
+    path = os.path.join(test_kegd_files_path, name)
+    if not os.path.exists(path):
+        raise ValueError(f'Path does not exists: {path}')
+    return KegDeploymentStrategyFiles(os.path.join(test_kegd_files_path, name))
+
+strategy_reader = kegd_model.DeploymentStrategyFileReader(
+    KegDeploymentProperties(),
+    None,
+    kegd_model.DeploymentStrategyParser()
+)
+
+def parse_strategy(strategy_file):
+    return strategy_reader.read(strategy_file, {})
+
+render_context_builder = ExtendedResourceTemplateContext(NameManager())
+
+def generate_render_context():
+    system_properties = {
+        'resourceId': '123',
+        'resourceName': 'just-testing',
+    }
+    resource_properties = {
+        'propertyA': 'A property',
+        'propertyB': 123
+    }
```

Comment:
Again, perhaps include non-primitive types like map or list here?

Reply:
I guess this method is a bit misleading as it's just a basic render context used in the tests. You'll see in `test_property_inputs_to_scripts` that we add properties of each supported type to this base context, to cover each one.
I've renamed this method, to help, but also I had some of the test names the wrong way round which doesn't help - so fixed those too!

Reply:
Ok, yeah I missed that. Should have looked closer. Looks good.